tests/drivers/dac/dac_loopback/src/test_dac.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2020 Libre Solar Technologies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * @addtogroup test_dac_loopback
+ * @{
+ * @defgroup t_dac_basic_loopback test_dac_loopback
+ * @brief TestPurpose: read back DAC driver output with ADC
+ * @}
+ */
+
+#include <drivers/dac.h>
+#include <drivers/adc.h>
+#include <zephyr.h>
+#include <ztest.h>
+
+/*
+ * We need to define an ADC channel to read back the output generated by the
+ * ADC. The two pins need to be connected with a jumper in order to pass the
+ * test in actual hardware.
+ */
+
+#if defined(CONFIG_BOARD_NUCLEO_L073RZ)
+
+/* DAC output on PA4, ADC input read from PA1 */
+
+#define DAC_DEVICE_NAME		DT_ALIAS_DAC1_LABEL
+#define DAC_RESOLUTION		12
+#define DAC_REFERENCE		DAC_REF_DEFAULT
+#define DAC_1ST_CHANNEL_ID	1
+
+#define ADC_DEVICE_NAME		DT_ADC_1_NAME
+#define ADC_RESOLUTION		12
+#define ADC_GAIN		ADC_GAIN_1
+#define ADC_REFERENCE		ADC_REF_INTERNAL
+#define ADC_ACQUISITION_TIME	ADC_ACQ_TIME_DEFAULT
+#define ADC_CHANNEL_ID		1
+
+#else
+#error "Unsupported board."
+#endif
+
+static const struct dac_channel_cfg dac_ch1_cfg = {
+	.channel_id = DAC_1ST_CHANNEL_ID,
+	.reference  = DAC_REFERENCE,
+	.buffered   = true
+};
+
+static const struct adc_channel_cfg adc_ch_cfg = {
+	.gain             = ADC_GAIN,
+	.reference        = ADC_REFERENCE,
+	.acquisition_time = ADC_ACQUISITION_TIME,
+	.channel_id       = ADC_CHANNEL_ID,
+};
+
+static struct device *init_dac(void)
+{
+	int ret;
+	struct device *dac_dev = device_get_binding(DAC_DEVICE_NAME);
+
+	zassert_not_null(dac_dev, "Cannot get DAC device");
+
+	ret = dac_channel_setup(dac_dev, &dac_ch1_cfg);
+	zassert_equal(ret, 0,
+		"Setting up of the first channel failed with code %d", ret);
+
+	return dac_dev;
+}
+
+/* ADC necessary to read back the value from DAC */
+static struct device *init_adc(void)
+{
+	int ret;
+	struct device *adc_dev = device_get_binding(ADC_DEVICE_NAME);
+
+	zassert_not_null(adc_dev, "Cannot get ADC device");
+
+	ret = adc_channel_setup(adc_dev, &adc_ch_cfg);
+	zassert_equal(ret, 0,
+		"Setting up of the ADC channel failed with code %d", ret);
+
+	return adc_dev;
+}
+
+/*
+ * test_dac_loopback
+ */
+static int test_task_loopback(void)
+{
+	int ret;
+
+	struct device *dac_dev = init_dac();
+	struct device *adc_dev = init_adc();
+
+	if (!dac_dev || !adc_dev) {
+		return TC_FAIL;
+	}
+
+	ret = dac_write_value(dac_dev, DAC_1ST_CHANNEL_ID, 2048, 12);
+	zassert_equal(ret, 0, "dac_write_value() failed with code %d", ret);
+
+	/* wait to let DAC output settle */
+	k_sleep(10);
+
+	static s16_t m_sample_buffer[1];
+	struct adc_sequence sequence = {
```

Comment:
Make this `static const struct add_sequence ...` to ensure non-specified members are set to zero.